bint.go

```diff
@@ -210,48 +210,160 @@ func parseBintFromU128(s []byte) (bool, bint, uint8, error) {
 		coef u128
 		prec uint8
 	)
-	for ; pos < width; pos++ {
-		if s[pos] == '.' {
+
+	if len(s[pos:]) <= 19 {
+		coef, prec, err = parseSmallToU128(s[pos:])
+	} else {
+		coef, prec, err = parseLargeToU128(s[pos:])
+	}
+
+	if err == ErrInvalidFormat {
+		return neg, bint{}, 0, errInvalidFormat(s)
+	}
+
+	return neg, bint{u128: coef}, prec, err
+}
+
+func parseSmallToU128(s []byte) (u128, uint8, error) {
+	var (
+		coef uint64
+		prec uint8
+	)
+
+	for i := 0; i < len(s); i++ {
+		if s[i] == '.' {
 			// return err if we encounter the '.' more than once
 			if prec != 0 {
-				return false, bint{}, 0, errInvalidFormat(s)
+				return u128{}, 0, ErrInvalidFormat
 			}
 
 			// nolint: gosec
-			prec = uint8(width - pos - 1)
+			prec = uint8(len(s) - i - 1)
 
 			// prevent "123." or "-123."
 			if prec == 0 {
-				return false, bint{}, 0, errInvalidFormat(s)
+				return u128{}, 0, ErrInvalidFormat
 			}
 
 			if prec > defaultPrec {
-				return false, bint{}, 0, ErrPrecOutOfRange
+				return u128{}, 0, ErrPrecOutOfRange
 			}
 
 			continue
 		}
 
-		if s[pos] < '0' || s[pos] > '9' {
-			return false, bint{}, 0, errInvalidFormat(s)
+		if s[i] < '0' || s[i] > '9' {
+			return u128{}, 0, ErrInvalidFormat
 		}
 
-		coef, err = coef.Mul64(10)
-		if err != nil {
-			return false, bint{}, 0, err
+		coef = coef*10 + uint64(s[i]-'0')
+	}
+
+	if coef == 0 {
+		return u128{}, 0, nil
+	}
+
+	return u128{lo: coef}, prec, nil
+}
+
+func parseLargeToU128(s []byte) (u128, uint8, error) {
+	// find '.' position
+	l := len(s)
+	pos := -1
+
+	for i := 0; i < len(s); i++ {
+		if s[i] == '.' {
+			pos = i
 		}
+	}
+
+	if pos == 0 || pos == l-1 {
+		return u128{}, 0, ErrInvalidFormat
+	}
+
+	var prec uint8
+	if pos != -1 {
+		// nolint: gosec
```

Comment:
```suggestion
		//nolint:gosec
```

I suggest you enabling `nolintlint` linter

Reply:
I didn't know about this. Is it a standard? What's the difference between this and having space?

Reply:
Go comments are starting with a space

https://tip.golang.org/doc/comment

Comments starting without a leading space are considered as language directive

Said otherwise, you can consider the fact leading space would imply to be visible on godoc